src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/Authentication/RemoteAppAuthenticationService.cs

```diff
@@ -113,17 +113,24 @@ internal static void AddHeaders(IEnumerable<string> headersToForward, HttpReques
 
         foreach (var headerName in headerNames)
         {
-            // Workaround for an issue identified by https://github.com/dotnet/systemweb-adapters/issues/228.
-            // HttpClient wrongly uses comma (",") instead of semi-colon (";") as a separator for Cookie headers.
-            // To mitigate this, we concatenate them manually and put them back as a single header value.
-            // This workaround can be removed once we target .NET 7+ as Kestrel is fixed there.
+            var originalHeaders = originalRequest.Headers[headerName].Where(x => !string.IsNullOrEmpty(x)).ToArray();

```

Comment:
Why split this up? I thought the issue was an empty header itself

Reply:
I've added test cases and applied the fix to Authorization header only.